### ISSUE TITLE
[camera] Add macOS support

### ISSUE THREAD

**Issue by stuartmorgan** [flutter/flutter#41708]
Since laptops commonly have cameras built in, this would be potentially useful.

Blocked on #30717

**Comment by larryaasen** [comment]
Any idea when this will be implemented?

**Comment by larryaasen** [comment]
I was looking for an indication as to when this would be prioritized. Is this on the roadmap yet?

**Comment by stuartmorgan** [comment]
No; see https://github.com/flutter/flutter/wiki/Issue-hygiene#when-will-my-bug-be-fixed

**Comment by larryaasen** [comment]
> If the issue is labeled P5 or P6, we are not currently expecting to ever work on it

Since it was labeled P5 it seems that it won’t be worked on. Is there a different plan for desktop cameras?

**Comment by 6LACK-N1NJA** [comment]
I'm so need it. Please 🙏

**Comment by moham96** [comment]
This should have been implemented with flutter 3.0 since that release marked the stable support of desktops

**Comment by stuartmorgan** [comment]
> Am I to assume that there is no way I can access a macbook's web cam and microphone in my Flutter app?

[Platform channels](https://docs.flutter.dev/development/platform-integration/platform-channels) are available on macOS, so you access arbitrary native code on macOS just as you can on other platforms. This issue is about the [`camera` plugin](https://pub.dev/packages/camera), not what is generally possible with respect to camera access.

**Comment by adilexperience** [comment]
> I have a Flutter app that uses Agora video calling and it works perfectly fine on the iPad. But my employer needs a macOs app and for the life of me I cant find any way to access Macbook's web cam or microphone. Am I to assume that there is no way I can access a macbook's web cam and microphone in my Flutter app?

have you found any solution for this? i am also struggling to request permissions from flutter macos.

**Comment by sportmachine** [comment]
me, too, Macos won't allow the camera access, any idea? can someone help? it is a flag issue.

**Comment by rohanjariwala03** [comment]
Any update for Camera package rather than working with Platform Channels?

**Comment by erlangparasu** [comment]
any update?

**Comment by stuartmorgan** [comment]
https://github.com/stuartmorgan/packages/tree/macos-camera has some initial exploration; it compiles now, but doesn't actually work yet (the preview is blank, and taking a picture results in an `NSError`, so there are likely capture configuration issues).

**Comment by naedx** [comment]
I would also love to see MacOS support for the camera plugin.

The use of cameras is common across devices. In addition, it would also speed up development targetting ios/android/etc. by being able to run this functionality directly for the macos platform.

**Comment by Mairramer** [comment]
@cdeil Are you still working on this?

I'm currently building a PoC for camera effects on both iOS and Android, and while investigating it I ran into the lack of camera support on macOS. Your implementation looks promising, so I was curious whether it's still being actively developed.

**Comment by Mairramer** [comment]
Hello @stuartmorgan,

I'm currently working locally on adding macOS support to `camera_avfoundation` based on the previous experiments. To make this migration as safe and reviewable as possible, I'd like to propose splitting the work into three focused PRs instead of one large PR. This structure could also be useful if someone else decides to contribute the same effort in the future.

The goal is to preserve Git history where possible and isolate risk:

PR 1: Directory Rename Only (`ios` → `darwin`)

This PR would contain only the directory move (`git mv ios darwin`) and the corresponding `sharedDarwinSource: true` configuration change, with no functional code changes.

PR 2: iOS Platform/UI Abstractions

This PR would introduce the necessary `#if os(iOS)` guards to isolate UIKit-specific dependencies (such as `UIDeviceOrientation`) from the core AVFoundation logic. It could be validated entirely on iOS, ensuring that existing behavior remains unchanged and minimizing the risk of regressions.

PR 3: macOS Implementation

The final PR would add the macOS deployment configuration, camera entitlements, and macOS-specific handling for hardware rotation (`videoRotationAngle`). If any unexpected macOS issues arise, this PR could be reverted independently without affecting the directory modernization introduced in the previous PRs.

WDYT?

**Comment by stuartmorgan-g** [comment]
As established in the comments just above, what you labelled as PR 1 must be a separate PR. PR 2 sounds fine as well.

As I said above, whether what you describe as PR 3 should actually be one PR or multiple PRs depends on how large that is, and how separable the parts are.

**Comment by cdeil** [comment]
@Mairramer - thanks for taking this on! I didn't submit PRs for this since I'm still learning about AVFoundation and Flutter and didn't feel that I met the bar set by the AI contribution guideline. My impression was that coding agents give a good solution directly here, but I wasn't sure e.g. if there's some non-obvious differences in MacOS vs iOS lifecycles or cases that would need to be handled.
